model/hex.py

```diff
@@ -0,0 +1,76 @@
+from typing import *
+from itertools import permutations
+
+from client.common import Hex as ResponseHex
+
+
+class Hex(NamedTuple):
+    q: int
+    r: int
+    s: int
+
+    def __add__(self, other):
+        return Hex(self.q + other.q, self.r + other.r, self.s + other.s)
+
+    def __sub__(self, other):
+        return Hex(self.q - other.q, self.r - other.r, self.s - other.s)
+
+    def distance(self, other=None):
+        # Returns the distance between two hexes.
+        #
+        # <param name="other">Other hex.</param>
+
+        if other is None:
+            other = Hex(0, 0, 0)
+
+        return sum(map(abs, self - other)) // 2
+
+    def neighbors(self, dist: int = 1):
+        # Returns the neighbors of the hex.
+        #
+        # <param name="dist">Distance from the hex to return neighbors at.</param>
+
+        for diff in hexes_at(dist):
+            yield self + diff
+
+    def range(self, *args):
+        # Returns the hexes in the given range.
+        #
+        # <param name="args">Range to return hexes in.</param>
+
+        for diff in hexes_range(*args):
+            yield self + diff
+
+    @staticmethod
+    def from_hex_response(hex: ResponseHex):
```

Comment:
I think it's better to make this method a classmethod, isn't it? The question is valid for all similar methods.